### PR TITLE
[menu-bar] Migrate FilePicker to Expo Modules

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Now that expo modules supports macOS, we can start to slowly migrate our modules to use the Sweet API

# How

Migrate FilePicker to use Expo Modules

# Test Plan

https://github.com/expo/orbit/assets/11707729/3ae9d671-97a7-4cdb-9bba-04061b7196d3

### CODE REVIEW

File: apps/menu-bar/modules/file-picker/ios/FilePickerModule.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+                    panel.allowedContentTypes = allowedTypes
+                }
+
+                if panel.runModal() == NSApplication.ModalResponse.OK {
+                    promise.resolve(panel.url?.path)
+                } else {
+                    promise.reject(Exception(name: "FILE_PICKER_ERROR", description: "NSModalResponseCancel"))
+                }
+            }
+        }
```
Comment (alanjhughes) [inline apps/menu-bar/modules/file-picker/ios/FilePickerModule.swift:45]:
```suggestion
        AsyncFunction("pickFileWithFilenameExtension"){  (filenameExtensions: [String], prompt: String) in
              NSApp.activate(ignoringOtherApps: true)
              let panel = NSOpenPanel()

              if !prompt.isEmpty {
                  panel.prompt = prompt
              }

              panel.allowsMultipleSelection = false
              panel.canChooseDirectories = true
              panel.canCreateDirectories = true

              if #available(macOS 11.0, *) {
                  var allowedTypes = [UTType]()

                  for extensionString in filenameExtensions {
                      let utTypesFromTag = UTType.types(tag: extensionString, tagClass: .filenameExtension, conformingTo: nil)
                      if !utTypesFromTag.isEmpty {
                          allowedTypes += utTypesFromTag
                      }


                      if let utTypeFromFilenameExtension = UTType(filenameExtension: extensionString), !allowedTypes.contains(utTypeFromFilenameExtension) {
                          allowedTypes.append(utTypeFromFilenameExtension)
                      }
                  } 

                  panel.allowedContentTypes = allowedTypes
              }

              guard panel.runModal() == NSApplication.ModalResponse.OK else {
                   throw Exception(name: "FILE_PICKER_ERROR", description: "NSModalResponseCancel")
              } 
              return panel.url?.path
        }.runOnQueue(.main)
```
We should just be able to run this on the main queue and remove the `Promise` parameter

File: apps/menu-bar/modules/file-picker/ios/FilePickerModule.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                panel.canChooseDirectories = true
+                panel.canCreateDirectories = true
+
+                if panel.runModal() == NSApplication.ModalResponse.OK {
+                    promise.resolve(panel.url?.path)
+                } else {
+                    promise.reject(Exception(name: "FILE_PICKER_ERROR", description: "NSModalResponseCancel"))
+                }
+            }
+        }
```
Comment (alanjhughes) [inline apps/menu-bar/modules/file-picker/ios/FilePickerModule.swift:61]: Same as above.

**Review by alanjhughes — CHANGES_REQUESTED** [review]
Great! I'll start helping out with this. Just a few comments

File: apps/menu-bar/modules/file-picker/ios/FilePickerModule.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import ExpoModulesCore
+import UniformTypeIdentifiers
+
+public class FilePickerModule: Module {
+    public func definition() -> ModuleDefinition {
+        Name("FilePicker")
+
+        AsyncFunction("pickFileWithFilenameExtension"){  (filenameExtensions: [String], prompt: String, promise: Promise) in
+            DispatchQueue.main.async {
```
Comment (tsapeta) [inline apps/menu-bar/modules/file-picker/ios/FilePickerModule.swift:9]: Sooo many spaces 🙉

**Review by alanjhughes — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]

**Pull request merged**